qiita_db/portal.py

```diff
@@ -5,13 +5,66 @@
 #
 # The full license is in the file LICENSE, distributed with this software.
 # -----------------------------------------------------------------------------
+import warnings
 
 from .sql_connection import SQLConnectionHandler
 from .util import convert_to_id
+from .exceptions import QiitaDBError, QiitaDBDuplicateError
 from qiita_core.qiita_settings import qiita_config
+from qiita_core.exceptions import IncompetentQiitaDeveloperError
 
 # Only make functions available for main portal
 if qiita_config.portal == "QIITA":
+    def create_portal(portal, desc):
```

Comment:
You lost me with `Transaction` and qiita_pet.

Regarding state, this is a very coarse check, but there are quite a few references to instance state:

```bash
$ grep -c self\. *.py
__init__.py:0
analysis.py:147
base.py:31
commands.py:0
data.py:129
environment_manager.py:0
exceptions.py:12
investigation.py:6
job.py:53
logger.py:19
meta_util.py:0
ontology.py:14
parameters.py:23
reference.py:10
search.py:38
sql_connection.py:62
study.py:98
user.py:44
util.py:0
```

Bringing this back on track, @squirrelo, my vote here is to make this a subclass of `QiitaObject` for consistency with the rest of the database interaction. I also vote against conditional imports. 

I'm not going to be responding to this thread further unless there are specific questions.